sssom/util.py

```diff
@@ -1505,3 +1512,84 @@ def _get_sssom_schema_object() -> SSSOMSchemaView:
         else SSSOMSchemaView()
     )
     return sssom_sv_object
+
+
+def flip_mappings(
+    df: pd.DataFrame, subject_prefix: str, merge_flipped: bool = True
+) -> pd.DataFrame:
+    """Switching subject and objects based on their prefixes and adjusting predicates accordingly.
+
+    :param df: Pandas dataframe.
+    :param subject_prefix: Prefix of subjects desired.
+    :param merge_flipped: If True (default), add flipped dataframe to input else,
+                          just return flipped data.
+    :return: Pandas dataframe with all subject IDs having the same prefix.
+    """
+    subject_starts_with_prefix_condition = df[SUBJECT_ID].str.startswith(
+        subject_prefix + ":"
+    )
+    object_starts_with_prefix_condition = df[OBJECT_ID].str.startswith(
+        subject_prefix + ":"
+    )
+    blank_predicate_modifier = df[PREDICATE_MODIFIER] == ""
+    predicate_flip_map = PREDICATE_FLIP_DICTIONARY
+
+    # predicate_modified_df = pd.DataFrame(df[not_predicate_modifier])
+    non_predicate_modified_df = pd.DataFrame(df[blank_predicate_modifier])
+
+    prefixed_subjects_df = pd.DataFrame(
+        non_predicate_modified_df[
+            (
+                subject_starts_with_prefix_condition
+                & ~object_starts_with_prefix_condition
+            )
+        ]
+    )
+    non_prefix_subjects_df = pd.DataFrame(
+        non_predicate_modified_df[
+            (
+                ~subject_starts_with_prefix_condition
+                & object_starts_with_prefix_condition
+            )
+        ]
+    )
+    df_to_flip = non_prefix_subjects_df.loc[
+        non_prefix_subjects_df[PREDICATE_ID].isin(list(predicate_flip_map.keys()))
+    ]
+    list_of_subject_object_columns = [
+        x for x in df_to_flip.columns if x.startswith(("subject", "object"))
```

Comment:
Sorry I didnt notice this before but can we reconcile this with:

https://github.com/mapping-commons/sssom-py/blob/master/sssom/parsers.py#L741

Reply:
I dont really know how applicable these are but I would kind of like if there was only one util method in sssom-py doing the flipping, rather than the same logic being at various places.

Reply:
This is for a `Mapping` object. What we have in this PR is for `MappingSetDataFrame`. I'll need to think about this.

Reply:
Can we make an issue to this end and resolve this comment?

Reply:
https://github.com/mapping-commons/sssom-py/issues/349